Yoast/wordpress-seo
# Only invalidate active taxonomies

## Summary

This PR can be summarized in the following changelog entry:
Only invalidate active taxonomies

## Test instructions

This PR can be tested by following these steps:

* Have a taxonomy that is not set as active in the Sitemap settings of the plugin.
* Save an object that uses this taxonomy.
* Make sure the type is not being triggered for clearing

Determining if the helper was triggered is a bit tricky.
The sitemaps are being cleared on 'shutdown'.

I have tested this by using temporary code to write data to a logfile:

In file: https://github.com/Yoast/wordpress-seo/blob/trunk/inc/sitemaps/class-sitemaps-cache.php#L232

Put the following code:
```php
file_put_contents( ABSPATH . '/sitemap-log-clear.txt', $_SERVER['REQUEST_TIME'] . PHP_EOL . var_export( $types, true ) . PHP_EOL, FILE_APPEND );
```

Fixes https://github.com/Yoast/wordpress-seo/issues/6318
